File: tests/fixtures/sample-js-project-positive-old/services/api-gateway/src/services/user.service.ts
import { config } from '../config';
import { formatUser } from '@sample/shared-utils';

interface UserData {
  name: string;
  email: string;
}

interface RawUser {
  id: string;
  name: string;
  email: string;
  createdAt: string;
}

interface FormattedUser {
  id: string;
  name: string;
  email: string;
  displayName: string;
  createdAt: string;
}

const TIMEOUT_MS = 10_000;

async function fetchJson(url: string): Promise<unknown> {
  const response = await fetch(url, { signal: AbortSignal.timeout(TIMEOUT_MS) });
  return response.json();
}

async function postJson(url: string, body: UserData): Promise<unknown> {
  const response = await fetch(url, {
    method: 'POST',
    headers: { 'Content-Type': 'application/json' },
    body: JSON.stringify(body),
    signal: AbortSignal.timeout(TIMEOUT_MS),
  });
  return response.json();
}

async function deleteUrl(url: string): Promise<void> {
  await fetch(url, { method: 'DELETE', signal: AbortSignal.timeout(TIMEOUT_MS) });
}

function toRawUser(data: unknown): RawUser {
  const obj = data as Record<string, unknown>;
  return {
    id: String(obj.id ?? ''),
    name: String(obj.name ?? ''),
    email: String(obj.email ?? ''),
    createdAt: String(obj.createdAt ?? ''),
  };
}

function toRawUserList(data: unknown): RawUser[] {
  const empty: RawUser[] = [];
  if (!Array.isArray(data)) return empty;
  return data.map(toRawUser);
}

export class UserService {
  private readonly baseUrl = config.userServiceUrl;

  async findAll(): Promise<FormattedUser[]> {
    const raw = await fetchJson(`${this.baseUrl}/users`);
    return toRawUserList(raw).map(formatUser);
  }

  async findById(id: string): Promise<FormattedUser> {
    const raw = await fetchJson(`${this.baseUrl}/users/${id}`);
    return formatUser(toRawUser(raw));
  }

  async create(userData: UserData): Promise<FormattedUser> {
    const raw = await postJson(`${this.baseUrl}/users`, userData);
    return formatUser(toRawUser(raw));
  }

  async handleRemove(id: string): Promise<void> {
    await deleteUrl(`${this.baseUrl}/users/${id}`);
  }
}

File: tests/fixtures/sample-csharp-project-negative/services/ApiGateway/Services/UserService.cs
using System.Text;
using System.Text.Json;
using Shared.Utils;

namespace ApiGateway.Services;

public class UserService
{
    private readonly HttpClient _httpClient;
    private readonly string _baseUrl;

    public UserService()
    {
        _httpClient = new HttpClient();
        _baseUrl = Environment.GetEnvironmentVariable("USER_SERVICE_URL") ?? "http://localhost:3001";
    }

    public List<Dictionary<string, object>> FindAll()
    {
        var response = _httpClient.GetAsync($"{_baseUrl}/users").Result;
        var json = response.Content.ReadAsStringAsync().Result;
        var users = JsonSerializer.Deserialize<List<Dictionary<string, object>>>(json)!;
        return users.Select(u => Formatters.FormatUser(u)).ToList();
    }

    public Dictionary<string, object>? FindById(string userId)
    {
        var response = _httpClient.GetAsync($"{_baseUrl}/users/{userId}").Result;
        var json = response.Content.ReadAsStringAsync().Result;
        var user = JsonSerializer.Deserialize<Dictionary<string, object>>(json)!;
        return Formatters.FormatUser(user);
    }

    public Dictionary<string, object> Create(Dictionary<string, object> data)
    {
        var content = new StringContent(JsonSerializer.Serialize(data), Encoding.UTF8, "application/json");
        var response = _httpClient.PostAsync($"{_baseUrl}/users", content).Result;
        var json = response.Content.ReadAsStringAsync().Result;
        var user = JsonSerializer.Deserialize<Dictionary<string, object>>(json)!;
        return Formatters.FormatUser(user);
    }

    public void Delete(string userId)
    {
        _httpClient.DeleteAsync($"{_baseUrl}/users/{userId}").Wait();
    }
}
